src/modules/requirements/TimeRequirement.ts

```diff
@@ -1,33 +1,26 @@
 import { AchievementOption } from '../GameConstants';
 import GameHelper from '../GameHelper';
 import Requirement from './Requirement';
+import DayCyclePart from '../dayCycle/DayCyclePart';
+import DayCycle from '../dayCycle/DayCycle';
 
 export default class TimeRequirement extends Requirement {
     private updateTrigger = ko.observable(0);
-    constructor(public startHour: number, public endHour: number, option = AchievementOption.more) {
+    constructor(public dayCycleParts: DayCyclePart[], option = AchievementOption.more) {
```

Comment:
Rename to `DayCyclePartRequirement`

Reply:
This was talking about the class name, sorry I wasn't clear. It doesn't take time inputs anymore, we are restricting by DayCycleParts, so the class name should reflect that

Reply:
that's what i thought at first, but then thought the line was meaningful, like for timeRestrict :^^ not easy to place that in a review : )